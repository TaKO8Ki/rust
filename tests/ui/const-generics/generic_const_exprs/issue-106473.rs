#![feature(generic_const_exprs)]
#![allow(incomplete_features)]

const DEFAULT: u32 = 1;

struct V<const U: usize = DEFAULT> //~ ERROR mismatched types
where
    [(); U]:;

trait Tr {}

impl Tr for V {}

fn main() {}
