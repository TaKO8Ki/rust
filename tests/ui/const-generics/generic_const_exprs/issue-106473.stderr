error[E0308]: mismatched types
  --> $DIR/issue-106473.rs:6:27
   |
LL | struct V<const U: usize = DEFAULT>
   |                           ^^^^^^^ expected `usize`, found `u32`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
